azurerm/data_source_batch_pool.go

```diff
@@ -274,6 +288,17 @@ func dataSourceArmBatchPoolRead(d *schema.ResourceData, meta interface{}) error
 			}
 		}
 
+		if props.DeploymentConfiguration != nil &&
```

Comment:
Same comment as below in the resource. we should nest a bunch of assignments and if's

Reply:
Cool, thanks for the comment, looks much batter. I made the update and slightly refactored the later code since it also uses `dcfg.VirtualMachineConfiguration`.